cachers/disk.go

```diff
@@ -0,0 +1,132 @@
+package cachers
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// indexEntry is the metadata that SimpleDiskCache stores on disk for an ActionID.
+type indexEntry struct {
+	Version   int    `json:"v"`
+	OutputID  string `json:"o"`
+	Size      int64  `json:"n"`
+	TimeNanos int64  `json:"t"`
+}
+
+// SimpleDiskCache is a LocalCache that stores data on disk.
+type SimpleDiskCache struct {
+	dir     string
+	verbose bool
+}
+
+func (dc *SimpleDiskCache) Kind() string {
+	return "disk"
+}
+
+func NewSimpleDiskCache(verbose bool, dir string) *SimpleDiskCache {
+	return &SimpleDiskCache{
+		dir:     dir,
+		verbose: verbose,
+	}
+}
+
+var _ LocalCache = &SimpleDiskCache{}
+
+func (dc *SimpleDiskCache) Start() error {
+	log.Printf("[%s]\tlocal cache in  %s", dc.Kind(), dc.dir)
+	return nil
+}
+
+func (dc *SimpleDiskCache) Close() error {
+	return nil
+}
+
+func (dc *SimpleDiskCache) Get(ctx context.Context, actionID string) (outputID, diskPath string, err error) {
+	actionFile := filepath.Join(dc.dir, fmt.Sprintf("a-%s", actionID))
+	ij, err := os.ReadFile(actionFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = nil
+			// if dc.verbose {
+			// 	log.Printf("disk miss: %v", actionID)
+			// }
+		}
+		return "", "", err
+	}
+	var ie indexEntry
+	if err := json.Unmarshal(ij, &ie); err != nil {
+		log.Printf("Warning: JSON error for action %q: %v", actionID, err)
```

Comment:
not sure if it's a thing but if so:
```suggestion
		log.Printf("Warning: JSON error for action %q: %w", actionID, err)
```

Reply:
%w can only be used in Errorf